From 8635d425c17498266e7ef6c0c518b1011d4c656d Mon Sep 17 00:00:00 2001
From: Lauri Tirkkonen <lotheac@iki.fi>
Date: Fri, 22 Jan 2016 18:48:50 +0200
Subject: [PATCH 2/6] make parallel make work on illumos

---
 getarch.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/getarch.c b/getarch.c
index 0a49fd1..ea6edbf 100644
--- a/getarch.c
+++ b/getarch.c
@@ -86,7 +86,7 @@ USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #include <sys/types.h>
 #include <sys/sysctl.h>
 #endif
-#ifdef linux
+#if defined(linux) || defined(__sun__)
 #include <sys/sysinfo.h>
 #include <unistd.h>
 #endif
@@ -892,7 +892,7 @@ static int get_num_cores(void) {
   size_t len;
 #endif
 
-#ifdef linux
+#if defined(linux) || defined(__sun__)
   //returns the number of processors which are currently online
   return sysconf(_SC_NPROCESSORS_ONLN);
 
-- 
2.6.1

