From 31aff441cea3f82694a07d4461c05e5ab1e978ed Mon Sep 17 00:00:00 2001
From: Lauri Tirkkonen <lotheac@iki.fi>
Date: Fri, 22 Jan 2016 18:50:53 +0200
Subject: [PATCH 4/6] use $(AWK) in Makefile.install and switch it to nawk

---
 Makefile.install | 2 +-
 Makefile.system  | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile.install b/Makefile.install
index 9814302..25783e3 100644
--- a/Makefile.install
+++ b/Makefile.install
@@ -29,7 +29,7 @@ install : 	lib.grd
 #for inc
 	@echo \#ifndef OPENBLAS_CONFIG_H > $(DESTDIR)$(OPENBLAS_INCLUDE_DIR)/openblas_config.h
 	@echo \#define OPENBLAS_CONFIG_H >> $(DESTDIR)$(OPENBLAS_INCLUDE_DIR)/openblas_config.h
-	@awk 'NF {print $$1, "OPENBLAS_"$$2, $$3}' config_last.h >> $(DESTDIR)$(OPENBLAS_INCLUDE_DIR)/openblas_config.h
+	@$(AWK) 'NF {print $$1, "OPENBLAS_"$$2, $$3}' config_last.h >> $(DESTDIR)$(OPENBLAS_INCLUDE_DIR)/openblas_config.h
 	@echo \#define OPENBLAS_VERSION \" OpenBLAS $(VERSION) \" >> $(DESTDIR)$(OPENBLAS_INCLUDE_DIR)/openblas_config.h
 	@cat openblas_config_template.h >> $(DESTDIR)$(OPENBLAS_INCLUDE_DIR)/openblas_config.h
 	@echo \#endif  \/\* OPENBLAS_CONFIG_H \*\/ >> $(DESTDIR)$(OPENBLAS_INCLUDE_DIR)/openblas_config.h
diff --git a/Makefile.system b/Makefile.system
index 42ad498..44e6be2 100644
--- a/Makefile.system
+++ b/Makefile.system
@@ -962,7 +962,7 @@ ifndef MD5SUM
 MD5SUM	= md5sum
 endif
 
-AWK	= awk
+AWK	= nawk
 
 REVISION = -r$(VERSION)
 MAJOR_VERSION = $(word 1,$(subst ., ,$(VERSION)))
-- 
2.6.1

