From 7ba1d9b9ca6958f1ad5f57d5ff57f1615d97dd5a Mon Sep 17 00:00:00 2001
From: Lauri Tirkkonen <lotheac@iki.fi>
Date: Fri, 22 Jan 2016 18:50:29 +0200
Subject: [PATCH 5/6] actually build the shared lib on illumos

---
 Makefile         | 2 +-
 exports/Makefile | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index 6ad87d8..35e01a7 100644
--- a/Makefile
+++ b/Makefile
@@ -85,7 +85,7 @@ endif
 
 shared :
 ifndef NO_SHARED
-ifeq ($(OSNAME), Linux)
+ifeq ($(OSNAME), $(filter $(OSNAME),Linux SunOS))
 	@$(MAKE) -C exports so
 	@-ln -fs $(LIBSONAME) $(LIBPREFIX).so
 	@-ln -fs $(LIBSONAME) $(LIBPREFIX).so.$(MAJOR_VERSION)
diff --git a/exports/Makefile b/exports/Makefile
index 177e975..2b3cdc4 100644
--- a/exports/Makefile
+++ b/exports/Makefile
@@ -110,7 +110,7 @@ endif
 dllinit.$(SUFFIX) : dllinit.c
 	$(CC) $(CFLAGS) -c -o $(@F) -s $<
 
-ifeq ($(OSNAME), Linux)
+ifeq ($(OSNAME), $(filter $(OSNAME),Linux SunOS))
 
 so : ../$(LIBSONAME)
 
-- 
2.6.1

