diff --git a/filter/foomatic-rip/process.c b/filter/foomatic-rip/process.c
index d93d2a0..3b9e718 100644
--- a/filter/foomatic-rip/process.c
+++ b/filter/foomatic-rip/process.c
@@ -90,7 +90,7 @@ void kill_all_processes()
     clear_proc_list();
 }
 
-static int _start_process(const char *name,
+static pid_t _start_process(const char *name,
                           int (*proc_func)(FILE *, FILE *, void *),
                           void *user_arg, FILE **pipe_in, FILE **pipe_out,
                           int createprocessgroup)
@@ -184,12 +184,12 @@ int exec_command(FILE *in, FILE *out, void *cmd)
     return EXIT_PRNERR_NORETRY_BAD_SETTINGS;
 }
 
-int start_system_process(const char *name, const char *command, FILE **fdin, FILE **fdout)
+pid_t start_system_process(const char *name, const char *command, FILE **fdin, FILE **fdout)
 {
     return _start_process(name, exec_command, (void*)command, fdin, fdout, 1);
 }
 
-int start_process(const char *name, int (*proc_func)(FILE *, FILE *, void *), void *user_arg, FILE **fdin, FILE **fdout)
+pid_t start_process(const char *name, int (*proc_func)(FILE *, FILE *, void *), void *user_arg, FILE **fdin, FILE **fdout)
 {
     return _start_process(name, proc_func, user_arg, fdin, fdout, 0);
 }
