diff --git a/filter/gstoraster.c b/filter/gstoraster.c
index 6567843..03e427e 100644
--- a/filter/gstoraster.c
+++ b/filter/gstoraster.c
@@ -525,12 +525,6 @@ out:
   return icc_profile;
 }
 
-static void
-child_reaper (int signum)
-{
-  wait(NULL);
-}
-
 int
 main (int argc, char **argv, char *envp[])
 {
@@ -568,9 +562,6 @@ main (int argc, char **argv, char *envp[])
   sa.sa_handler = SIG_IGN;
   sigaction(SIGPIPE, &sa, NULL);
 
-  sa.sa_handler = child_reaper;
-  sigaction(SIGCHLD, &sa, NULL);
-
   num_options = cupsParseOptions(argv[5], 0, &options);
 
   t = getenv("PPD");
