diff --git a/src/keymap.c b/src/keymap.c
index 5554e8b..c9ce279 100644
--- a/src/keymap.c
+++ b/src/keymap.c
@@ -276,7 +276,7 @@ printascii(
 	if (ch == 0)
 		strcpy(buf, _("NULL"));
 #if defined(MULTIBYTE_ABLE) && !defined(NO_LOCALE)
-	else if (iswgraph(ch)) {	/* Regular printables */
+	else if (iswctype(ch, wctype("graph"))) {	/* Regular printables */
 		int i = wctomb(buf, ch);
 
 		if (i > 0)
