--- a/configure	Mon Jan 11 14:12:12 2016
+++ b/configure	Mon Jan 11 14:19:20 2016
@@ -392,7 +392,12 @@
   retval = ()
   for flag in ['--libs-only-l', '--cflags-only-I', '--libs-only-L']:
     try:
-      val = subprocess.check_output([pkg_config, args, flag, pkg])
+      cmd = [pkg_config, args, flag, pkg]
+      pipe = subprocess.Popen(cmd)
+      val, _ = proc.communicate()
+      if pipe.wait() != 0:
+          raise subprocess.CalledProcessError(returncode=pipe.returncode,
+                                              cmd=cmd)
       # check_output returns bytes
       val = val.encode().strip().rstrip('\n')
     except subprocess.CalledProcessError:
