diff --git a/include/config.h b/include/config.h
index 873f96a..a25e92d 100644
--- a/include/config.h
+++ b/include/config.h
@@ -143,10 +143,10 @@
 
 #ifndef WIZARD		/* allow for compile-time or Makefile changes */
 # ifndef KR1ED
-#  define WIZARD  "wizard" /* the person allowed to use the -D option */
+#  define WIZARD  "ltirkkon" /* the person allowed to use the -D option */
 # else
 #  define WIZARD
-#  define WIZARD_NAME "wizard"
+#  define WIZARD_NAME "ltirkkon"
 # endif
 #endif
 
@@ -348,10 +348,14 @@ typedef unsigned char	uchar;
  * bugs left here.
  */
 
+/* Enable the patches */
+#define HPMON
+#define PARANOID
+
 #if defined(TTY_GRAPHICS) || defined(MSWIN_GRAPHICS)
 # define MENU_COLOR
 # define MENU_COLOR_REGEX
-/*# define MENU_COLOR_REGEX_POSIX */
+# define MENU_COLOR_REGEX_POSIX
 /* if MENU_COLOR_REGEX is defined, use regular expressions (regex.h,
  * GNU specific functions by default, POSIX functions with
  * MENU_COLOR_REGEX_POSIX).
