--- a/include/config.h	Fri Jul 12 22:53:11 2013
+++ b/include/config.h	Fri Jul 12 22:54:18 2013
@@ -348,10 +348,14 @@
  * bugs left here.
  */
 
+/* Enable the patches */
+#define HPMON
+#define PARANOID
+
 #if defined(TTY_GRAPHICS) || defined(MSWIN_GRAPHICS)
 # define MENU_COLOR
 # define MENU_COLOR_REGEX
-/*# define MENU_COLOR_REGEX_POSIX */
+# define MENU_COLOR_REGEX_POSIX
 /* if MENU_COLOR_REGEX is defined, use regular expressions (regex.h,
  * GNU specific functions by default, POSIX functions with
  * MENU_COLOR_REGEX_POSIX).
