diff --git a/dat/yendor.des b/dat/yendor.des
index 34b7809..d0878a3 100644
--- a/dat/yendor.des
+++ b/dat/yendor.des
@@ -50,7 +50,7 @@ MONSTER:('@',"Wizard of Yendor"),(16,05),asleep
 MONSTER:('d',"hell hound"),(15,05)
 MONSTER:('V',"vampire lord"),(17,05)
 # The local treasure
-OBJECT:('+',"Book of the Dead"),(16,05)
+OBJECT:('+',"Master's Thesis"),(16,05)
 # Surrounding terror
 MONSTER:(';',"kraken"),(14,02)
 MONSTER:(';',"giant eel"),(17,02)
diff --git a/src/objects.c b/src/objects.c
index 3c68674..9ca28d0 100644
--- a/src/objects.c
+++ b/src/objects.c
@@ -987,7 +987,7 @@ OBJECT(OBJ("novel", "paperback"),
        BITS(0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, P_NONE, HI_PAPER),
        0, SPBOOK_CLASS, 0, 0, 0, 20, 0, 0, 0, 1, 20, CLR_BRIGHT_BLUE),
 /* a special, one of a kind, spellbook */
-OBJECT(OBJ("Book of the Dead", "papyrus"),
+OBJECT(OBJ("Master's Thesis", "papyrus"),
        BITS(0, 0, 1, 0, 1, 0, 1, 1, 0, 0, 0, P_NONE, PAPER),
        0, SPBOOK_CLASS, 0, 0, 20, 10000, 0, 0, 0, 7, 20, HI_PAPER),
 #undef SPELL
diff --git a/src/spell.c b/src/spell.c
index 71068c7..a5f0c98 100644
--- a/src/spell.c
+++ b/src/spell.c
@@ -206,7 +206,7 @@ struct obj *book2;
     struct monst *mtmp, *mtmp2;
     coord mm;
 
-    You("turn the pages of the Book of the Dead...");
+    You("turn the pages of the Master's Thesis...");
     makeknown(SPE_BOOK_OF_THE_DEAD);
     /* KMH -- Need ->known to avoid "_a_ Book of the Dead" */
     book2->known = 1;
diff --git a/src/write.c b/src/write.c
index 1000b29..d773641 100644
--- a/src/write.c
+++ b/src/write.c
@@ -213,7 +213,7 @@ found:
         pline("It's obscene!");
         return 1;
     } else if (i == SPE_BOOK_OF_THE_DEAD) {
-        pline("No mere dungeon adventurer could write that.");
+        pline("No mere phuksi could write that.");
         return 1;
     } else if (by_descr && paper->oclass == SPBOOK_CLASS
                && !objects[i].oc_name_known) {
diff --git a/util/makedefs.c b/util/makedefs.c
index 0e340b4..0285a4a 100644
--- a/util/makedefs.c
+++ b/util/makedefs.c
@@ -2514,6 +2514,7 @@ do_objs()
     }
     Fprintf(ofp, "%s", Dont_Edit_Code);
     Fprintf(ofp, "#ifndef ONAMES_H\n#define ONAMES_H\n\n");
+    Fprintf(ofp, "#define SPE_BOOK_OF_THE_DEAD SPE_MASTER_S_THESIS\n\n");
 
     for (i = 0; !i || objects[i].oc_class != ILLOBJ_CLASS; i++) {
         SpinCursor(3);
