--- a/configure	Wed Oct 23 13:59:38 2013
+++ b/configure	Wed Oct 23 13:59:50 2013
@@ -14350,18 +14350,8 @@
 done
 
 
-for ac_func in posix_fallocate
-do :
-  ac_fn_c_check_func "$LINENO" "posix_fallocate" "ac_cv_func_posix_fallocate"
-if test "x$ac_cv_func_posix_fallocate" = xyes; then :
-  cat >>confdefs.h <<_ACEOF
-#define HAVE_POSIX_FALLOCATE 1
-_ACEOF
 
-fi
-done
 
-
 
 case $TERM in
        #   for the most important terminal types we directly know the sequences
