From 52d46bc23ac5e28e8fb8639db85ec88d8965f470 Mon Sep 17 00:00:00 2001
From: Lauri Tirkkonen <lotheac@iki.fi>
Date: Mon, 6 Oct 2014 16:06:10 +0300
Subject: [PATCH] let smf manage pids instead of using a pidfile

---
 init/znapzend.xml.in | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/init/znapzend.xml.in b/init/znapzend.xml.in
index c27930a..54764bc 100644
--- a/init/znapzend.xml.in
+++ b/init/znapzend.xml.in
@@ -28,13 +28,13 @@
         <exec_method 
                 type="method" 
                 name="start" 
-                exec="@prefix@/bin/znapzend --daemonize --pidfile=/var/run/znapzend.pid" 
+                exec="@prefix@/bin/znapzend --daemonize --pidfile=/dev/null"
                 timeout_seconds="170" />
 
         <exec_method
                 type="method"
                 name="stop"
-                exec="/usr/bin/kill `cat /var/run/znapzend.pid`"
+                exec=":kill"
                 timeout_seconds="60" />
 
 
-- 
1.8.1.3

