From e93efbf63e89d9f5dddd6ced1d317acf1414494f Mon Sep 17 00:00:00 2001
From: Lauri Tirkkonen <lotheac@iki.fi>
Date: Mon, 6 Oct 2014 17:20:10 +0300
Subject: [PATCH 2/2] fix mandir and binary path in svc manifest

---
 init/znapzend.xml.in | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/init/znapzend.xml.in b/init/znapzend.xml.in
index 54764bc..e7c216a 100644
--- a/init/znapzend.xml.in
+++ b/init/znapzend.xml.in
@@ -28,7 +28,7 @@
         <exec_method 
                 type="method" 
                 name="start" 
-                exec="@prefix@/bin/znapzend --daemonize --pidfile=/dev/null"
+                exec="@exec_prefix@/bin/znapzend --daemonize --pidfile=/dev/null"
                 timeout_seconds="170" />
 
         <exec_method
@@ -48,9 +48,9 @@
                 </loctext>
             </common_name>
             <documentation>
-                <manpage title="znapzend" section="1" manpath="@prefix@/share/man" />
-                <manpage title="znapzendzetup" section="1" manpath="@prefix@/share/man" />
-                <manpage title="znapzendztatz" section="1" manpath="@prefix@/share/man" />
+                <manpage title="znapzend" section="1" manpath="@mandir@" />
+                <manpage title="znapzendzetup" section="1" manpath="@mandir@" />
+                <manpage title="znapzendztatz" section="1" manpath="@mandir@" />
             </documentation>
         </template>
 </service>
-- 
1.9.1

