diff --git a/bin/znapzend b/bin/znapzend
index 7d4b0a1..c7a3c21 100755
--- a/bin/znapzend
+++ b/bin/znapzend
@@ -1,4 +1,4 @@
-#!/usr/bin/env perl
+#!/opt/niksula/perl5/bin/perl
 
 use FindBin;
 # PERL5LIB
diff --git a/bin/znapzendzetup b/bin/znapzendzetup
index ddc997c..0fcb3eb 100755
--- a/bin/znapzendzetup
+++ b/bin/znapzendzetup
@@ -1,4 +1,4 @@
-#!/usr/bin/env perl
+#!/opt/niksula/perl5/bin/perl
 
 use FindBin;
 # PERL5LIB
diff --git a/bin/znapzendztatz b/bin/znapzendztatz
index 38c004f..35be9d1 100755
--- a/bin/znapzendztatz
+++ b/bin/znapzendztatz
@@ -1,4 +1,4 @@
-#!/usr/bin/env perl
+#!/opt/niksula/perl5/bin/perl
 
 use FindBin;
 # PERL5LIB
