--- a/lib/ZnapZend/ZFS.pm	Tue Dec 16 18:08:31 2014
+++ b/lib/ZnapZend/ZFS.pm	Tue Mar 24 16:03:30 2015
@@ -16,7 +16,7 @@
 has connectTimeout  => sub { 30 };
 has propertyPrefix  => sub { q{org.znapzend} };
 has sshCmdArray     => sub { [qw(ssh -o Compression=yes -o CompressionLevel=1 -o),
-    qw(Cipher=arcfour -o batchMode=yes -o), 'ConnectTimeout=' . shift->connectTimeout] };
+    qw(batchMode=yes -o), 'ConnectTimeout=' . shift->connectTimeout] };
 has mbufferParam    => sub { [qw(-q -s 128k -W 60 -m)] }; #don't remove the -m as the buffer size will be added
 has scrubInProgress => sub { qr/scrub in progress/ };
 
