diff --git a/install.rb b/install.rb
index f8cbdc4..49acfb6 100755
--- a/install.rb
+++ b/install.rb
@@ -96,9 +96,6 @@ def do_man(man, strip = 'man/')
     FileUtils.makedirs(om, {:mode => 0755, :verbose => true})
     FileUtils.chmod(0755, om)
     FileUtils.install(mf, omf, {:mode => 0644, :preserve => true, :verbose => true})
-    gzip = %x{which gzip}
-    gzip.chomp!
-    %x{#{gzip} -f #{omf}}
   end
   else
   puts "Skipping Man Page Generation"
